### PR TITLE
Add modal that signifies location is being found and add autocomplete/reverse geocoding to address input

### LINKED ISSUE #50: Geolocation takes a couple of seconds to respond when user wants to use their location

### LINKED ISSUE #55: Autocomplete addresses
Use Google Maps API autocomplete service to autocomplete typed in addresses.

### PR DESCRIPTION
**Pull request by PriyaVarra**
This pull request adds a few more features to improve user interaction.

1. If the geolocation request for obtaining the location takes more than 250 ms, a modal pops up letting the user know that their location is still being found, and then it is closed once the location is found. I did this to avoid the modal popping up after their location is found when the response is fast.

2. The address input field now has autocomplete capabilities and the drawer width was increased to allow a better view of the addresses as seen below.

.

3. The buttons for "Use My Location" and "Use Address" have both been moved under the address input field as seen below
.
Additionally, "Use My Location" reverse geocodes the obtained latitude and longitude and populates the address input field with the corresponding result.

### CODE REVIEW

File: where-to-travel/src/main/webapp/style.css [hunk 1]
```diff
@@ -110,6 +117,11 @@ html {
...
+.text-center {
+  text-align: center;
+}
+
 .text-color {
   color: #1E3C19;
-}
\ No newline at end of file
+}
+
```
Comment (andy7809) [inline where-to-travel/src/main/webapp/style.css]: remove extra line at eof

**Review by andy7809 — COMMENTED** [review]
One extra line, otherwise looks great!

File: where-to-travel/src/main/webapp/script.js [hunk 2]
```diff
@@ -57,6 +57,11 @@ let focusedPin;
 let homeMarker = null;
 let markers = [];
 
+// Flage indicating if geolocation is running
```
Comment (emmapaczkowski) [inline where-to-travel/src/main/webapp/script.js]: Did you mean to say flag?

Comment (PriyaVarra) [reply where-to-travel/src/main/webapp/script.js]: lol yes. fixed!

File: where-to-travel/src/main/webapp/style.css (hunk 1 above)
Comment (PriyaVarra) [reply where-to-travel/src/main/webapp/style.css]: fixed!

File: where-to-travel/src/main/webapp/style.css [hunk 3]
```diff
@@ -6,6 +6,14 @@
   z-index: 500;
 }
 
+.bmd-drawer-f-l>.bmd-layout-drawer {
+  top: 0;
+  left: 0;
+  width: 360px;
```
Comment (wjiamin) [inline where-to-travel/src/main/webapp/style.css]: alphabetize declarations?

File: where-to-travel/src/main/webapp/script.js [hunk 4]
```diff
@@ -106,23 +117,55 @@ function showInfoModal() {
  *
  * @param {boolean} useAddress Flag indicating whether to get location from browser or address
  */
-function getHomeLocation(useAddress) {
-  let locationFunction = () => getLocationFromBrowser();
+function getHomeLocation(useAddress) { 
+  const addressInput = document.getElementById('addressInput');
 
+  let locationFunction = () => getLocationFromBrowser();
```
Comment (wjiamin) [inline where-to-travel/src/main/webapp/script.js]: Place this in an else function instead? To prevent this being viewed as default.

**Review by wjiamin — COMMENTED** [review]
LGTM, I'll let Sailesh approve.

File: where-to-travel/src/main/webapp/style.css (hunk 3 above)
Comment (PriyaVarra) [reply where-to-travel/src/main/webapp/style.css]: fixed!

File: where-to-travel/src/main/webapp/script.js (hunk 4 above)
Comment (PriyaVarra) [reply where-to-travel/src/main/webapp/script.js]: fixed!

**Review by saileshsimhadri — APPROVED** [review]

**Pull request merged**